lib/Doctrine/ODM/CouchDB/Proxy/ProxyFactory.php

```diff
@@ -187,12 +188,14 @@ private function generateMethods(ClassMetadata $class)
                         $argumentString  .= ', ';
                     }
 
-                    // We need to pick the type hint class too
-                    if (($paramClass = $param->getClass()) !== null) {
-                        $parameterString .= '\\' . $paramClass->getName() . ' ';
-                    } else if ($param->isArray()) {
-                        $parameterString .= 'array ';
-                    }
+	                // We need to pick the type hint class too
+	                if (($paramClass = $param->getClass()) !== null) {
```

Comment:
This code should rather pint at the `doctrine/common` proxy-factory, or use the ProxyManager component, like is done in https://github.com/doctrine/orm/pull/6719

Reply:
I pushed a new commit to use the new ProxyManager component. Did some tests to help travis run go smoothly, but it's not very efficient. Any comment on the new code ?
I did not started a full code review/refactoring as it's working, but I saw a lot a problems that need to be addressed.